debate_app.py

```diff
@@ -1,128 +1,314 @@
+                    "\n[yellow]Would you like to try entering the API keys again?[/yellow]",
+                    choices=["y", "n"]
+                )
+                if retry.lower() != "y":
+                    raise SystemExit("Exiting due to API key validation failure")
+        self.judge: Optional[LLMPersonality] = None
+        
+    def _check_and_request_api_keys(self) -> None:
+        """Check for required API keys and prompt user if any are missing.
+        
+        This method validates the presence of API keys for selected providers
+        and interactively requests them if they're not found in the environment.
+        
+        Raises:
+            ValueError: If an API key is invalid or missing and user chooses not to provide it.
+        """
+        missing_keys: List[Tuple[str, str]] = []
+        if "claude" in self.selected_providers and not os.getenv("CLAUDE_API_KEY"):
+            missing_keys.append(("CLAUDE_API_KEY", "Claude (Anthropic)"))
+        if "openai" in self.selected_providers and not os.getenv("OPENAI_API_KEY"):
+            missing_keys.append(("OPENAI_API_KEY", "OpenAI"))
+        if "gemini" in self.selected_providers and not os.getenv("GEMINI_API_KEY"):
+            missing_keys.append(("GEMINI_API_KEY", "Gemini (Google)"))
+            
+        if missing_keys:
+            console.print("\n[bold red]Missing API Keys[/bold red]")
+            console.print("To proceed, you'll need to provide API keys for the selected providers.")
+            console.print("You can get them from:")
+            console.print("- Claude: https://console.anthropic.com/")
+            console.print("- OpenAI: https://platform.openai.com/api-keys")
+            console.print("- Gemini: https://aistudio.google.com/apikey\n")
+            
+            for env_var, provider in missing_keys:
+                key = Prompt.ask(f"[yellow]Enter your {provider} API Key[/yellow]")
+                if not key.strip():
+                    raise ValueError(f"API key for {provider} is required")
+                os.environ[env_var] = key.strip()
+            
+            console.print("\n[green]API keys have been temporarily set for this session.[/green]")
+            console.print("[yellow]Tip: Add these to your .env file to avoid entering them again![/yellow]\n")
+    
+    def _select_providers(self) -> List[str]:
+        """Let user select which AI providers to use in the debate.
+        
+        Returns:
+            List[str]: List of selected provider names ('claude', 'openai', 'gemini')
+        """
+        console.print("\n[bold cyan]Select AI Providers for the Debate:[/bold cyan]")
+        console.print("[1] Claude (Anthropic) only")
+        console.print("[2] OpenAI only")
+        console.print("[3] Gemini (Google) only")
+        console.print("[4] Claude + OpenAI")
+        console.print("[5] Claude + Gemini")
+        console.print("[6] OpenAI + Gemini")
+        console.print("[7] All providers")
```

Comment:
What do you think of letting users select multiple? Just list the options and let users enable/disable individually